tests/test_s3.py

```diff
@@ -1543,6 +1543,164 @@ def test_has_statement_policy(self):
         resources = p.run()
         self.assertEqual(len(resources), 1)
 
+    def test_bucket_replication_policy_remove(self):
+        replicated_from_name = "replication-from-12345"
+        replicated_to_name = "replication-t0-12345"
+
+        self.patch(s3.S3, "executor_factory", MainThreadExecutor)
+        self.patch(
+            s3.SetBucketReplicationConfig, "executor_factory", MainThreadExecutor
+        )
+
+        # Remove GetBucketLocation from augments
+        # was causing AttributeError: 'FakeHttpResponse' object has no attribute 'content'
+        augments = list(s3.S3_AUGMENT_TABLE)
+        augments.remove((
+            "get_bucket_location", "Location", {}, None, 's3:GetBucketLocation'))
+        self.patch(s3, "S3_AUGMENT_TABLE", augments)
+
+        # and ignore any other buckets we might have in this test account
+        # to minimize the placebo data and API calls
+        # Inspired by #5206, thanks tjstansell!
+        self.patch(
+            s3.S3.resource_type,
+            "enum_spec",
+            ('list_buckets', "Buckets[?Name=='{}']".format(replicated_from_name), None)
+        )
+        session_factory = self.replay_flight_data("test_s3_replication_policy_remove")
+        session = session_factory()
+        client = session.client("s3")
+        account_id = session.client('sts').get_caller_identity().get('Account')
+        self.addCleanup(destroyBucket, client, replicated_from_name)
+        self.addCleanup(destroyBucket, client, replicated_to_name)
+
+        # Setup buckets for replication by creating and adding versioning
+        for bucket in (replicated_from_name, replicated_to_name):
+            client.create_bucket(
+                Bucket=bucket
+            )
+            client.put_bucket_versioning(
+                Bucket=bucket,
+                VersioningConfiguration={
+                    'MFADelete': 'Disabled',
+                    'Status': 'Enabled'
+                }
+            )
+
+        # add a replication policy between them
+        client.put_bucket_replication(
+            Bucket=replicated_from_name,
+            ReplicationConfiguration={
+                'Role': 'arn:aws:iam::' + account_id + ':role/s3_replicator',
+                'Rules': [
+                    {
+                        'ID': 'test12345',
+                        'Prefix': '',
+                        'Status': 'Enabled',
+                        'Destination': {
+                            'Bucket': 'arn:aws:s3:::' + replicated_to_name
+                        }
+                    }
+                ]
+            }
+        )
+
+        # Test to make sure that a replication policy was in fact created between the buckets
+        response = client.get_bucket_replication(Bucket=replicated_from_name)
+        for rule in response['ReplicationConfiguration']['Rules']:
+            self.assertEqual(rule['Status'], 'Enabled')
+
+        p = self.load_policy(
```

Comment:
the setup above feels extraneous, its useful to a fully self contained functional test, but we're in process on improving that, sans a functional test setup the infra setup outside the test, use a filter to validate pre-conditions, modify in action, use a client to verify actions took effect.

Reply:
4b8291831aae806235731e76a1928dbfb915723a I agree this is more of a functional setup. I have noticed that sometimes I have trouble drawing the line on these tests.

Reply:
there's several hundred lines of recorded data, and extra api calls, in addition to 50 lines of test that can be removed, its not marked functional, and as previously noted, there's work in progress to make this independent of procedural api calls in test code and recorded responses.

Reply:
I'll trim out as much as I can. Because there is work in progress to make the tests independent of procedural api calls and recorded responses? I can patch this in my local version but wanted to submit for others to use as well.

Reply:
trim 181 lines, reduce to 850 from 1031 3b67095fcba7d4aeac05c9ed0e3b70cb70d7b1eb

Reply:
I would just omit, the extraneous bits here (creation, setup & destruction of buckets) and the recorded responses for the same. ie. setup the infra setup outside the test, use a filter to validate pre-conditions, modify in action, use a client to verify actions took effect.

Reply:
Ah, I think I hear what you are saying this time. We would never really need to regenerate those recordings, right? So just test that the function mutates existing infra properly.  I can see where that would cut down. More reductions to come :)

Reply:
Here you go my friend! 49ae217df29baf309a7494e013e0c29e8a82cd6b trims to 515 so literally a 50% reduction. I had no idea how bloated the PR was before. Is this getting close to lean?

Reply:
so for tests that we do want to re-record we use the functional test decorator, this varies somewhat by provider (in azure all tests are re-recordable and executed functionally on a nightly cron, but none of the provisioning is in the test code/recorded responses instead it uses arm templates (akin to cfn). as i noted, this is going to change for the aws provider to match that, as we don't want to manually have test code around provisioning/extraneous test data as its more laborious on the dev, ci times, etc.

re lean, see #5452 for a clean up script, most of the response metadata can be omitted.

Reply:
Thanks Kapil! This is awesome! I can 100% see your thoughts on where the fat is now.  I may try to refactor and contribute as a housekeeping utility where it could be run on the whole project or just on specific test dirs.